RIPS/rip-9999.md

```diff
@@ -0,0 +1,313 @@
+any action that will increment its nonce.
+
+Transaction hash uniqueness is therefore guaranteed by a `bigNonce` field for the `NonceManager` contract.
+
+The main difference between `DELEGATED_EOA_SUBTYPE` and `AUTHORIZED_EOA_SUBTYPE` is that the transaction is handled
+as a full RIP-7560 Account Abstraction transaction.
+
+The `implementationAddress` will be invoked with the `validateTransaction` frame same as a regular
+Smart Contract Account would.
+This allows EOAs to temporarily transfer control of their assets to a Smart Contract Account but maintain an
+ability to revoke or override this authorisation.
+This also allows a cheap creation of AA accounts that do not require any code deployment, if the private key
+used to create the `yParity, r, s` signature does not exist.
+
+Note that most of the transaction fields are not signed by this transaction's signature.
+This means that the "implementation" contract is the entity responsible for implementing all kinds of protections,
+and the EOA account that signs an authorization fully cedes control to the "implementation" until
+the authorization is revoked.
+
+For the purposes of Gas Abstraction, the `paymasterData` parameter is not signed as well,
+so different Paymaster contracts can be used for subsequent authorized transactions using the same authorization.
+
+#### Using permanent authorizations to create Smart Contract Accounts
+
+This approach relies on the so-called "Nick's method", which includes deriving an address from some data and
+a randomized signature value, without ever knowing the ECDSA private key that would be able to control this address.
+
+Applying this approach to an `AUTHORIZED_EOA_SUBTYPE` transaction, a new account can be created,
+whose address is forever bound to `implementationAddress` and `commit`,
+without ever needing to deploy code for the contract.
+
+### Gas-sponsored time-limited code injection EOA transaction subtype
+
+```
+INSERT_CODE_EOA_SUBTYPE
+```
+
+[//]: # (TODO: 7377 probably needs codeAddr/storage simply because it has no deployment frame; we can have initCode)
+
+This transaction subtype extends the `SPONSORED_EOA_SUBTYPE` transactions with an ability to insert
+contract code in the context of the EOA sender.
+
+The transaction payload for this subtype is extended with the following fields:
+```
+callGasLimit,
+codeAddr,
+storage,
+data,
+value
+```
+
+[//]: # (TODO: TBD: this is copied directly from 7377 as it seems to apply well.)
+
+Unlike standard contract deployment, a migration transaction directly specifies what code value the sender's
+account should be set to.
+
+As the first step of processing the transaction, set the sender’s code to `state[tx.codeAddr].code`.
+Next, for each tuple in `storage` and the sender’s storage trie, set `storage[t.first] = t.second`.
+
+Now instantiate an EVM call into the sender's account using the same rules as EIP-1559 and set the
+transaction's origin to be `keccak256(sender)[0..20]`.
```

Comment:
> origin to be `keccak256(sender)[0..20]`

Do we want to copy this?  7377 did this to avoid having to deal with the issues raised by letting `tx.origin` have code execution.  The RIP-7560 family already does that anyway.  In addition, the delegation subtype above makes it possible for `tx.origin` to have code (similarly to 3074), so there's no need to have special treatment for ORIGIN in this subtype.